frontend/js/tests/user/import-listens/ImportListens.test.tsx

```diff
@@ -0,0 +1,164 @@
+    window.HTMLElement.prototype.scrollIntoView = jest.fn();
+
+    server = setupServer(
+      http.post("/settings/import/", () => {
+        return HttpResponse.json({ user_has_email: true });
+      }),
+      http.get("/1/import-listens/list/", () => {
+        return HttpResponse.json(mockImports);
+      }),
+      http.post("/1/import-listens/cancel/1/", () => {
+        return HttpResponse.json({ success: true, id: 1 });
+      }),
+      http.post("/1/import-listens/1/", () => {
+        return HttpResponse.json(mockImports[0]);
+      })
+    );
+    server.listen();
+
+    router = createMemoryRouter(routes, {
+      initialEntries: ["/settings/import/"],
+    });
+  });
+
+  afterAll(() => {
+    server.close();
+  });
+
+  it("renders submission form correctly", async () => {
+    renderWithProviders(
+      <RouterProvider router={router} />,
+      {},
+      { wrapper: ReactQueryWrapper },
+      false
+    );
+    await waitFor(() => {
+      expect(screen.getByText(/start import from/i)).toBeInTheDocument();
+      
+    });
+    await waitFor(() => {
+      expect(screen.getByText(/end date for import/i)).toBeInTheDocument();
+      
+    });
+    await waitFor(() => {
+      expect(screen.getByText(/select Service/i)).toBeInTheDocument();
+      
+    });
+    await waitFor(() => {
+      expect(screen.getByText(/choose a File/i)).toBeInTheDocument();
+      
+    });
+    await waitFor(() => {
+      expect(screen.getByRole("button", { name: /import listens/i })).toBeInTheDocument();
+      
+    });
+  });
+  
+  it("renders imports with correct data", async () => {
+    renderWithProviders(
+      <RouterProvider router={router} />,
+      {},
+      { wrapper: ReactQueryWrapper },
+      false
+    );
+    await waitFor(() => {
+      expect(screen.getByText("spotify.zip")).toBeInTheDocument();
+      
+    });
+    await waitFor(() => {
+      expect(screen.getByText("August 11th, 2025")).toBeInTheDocument();
+      
+    });
+    await waitFor(() => {
+      expect(screen.getByText("-")).toBeInTheDocument();
+      
+    });
+    
+      
+    
+  });
+
+  it("calls cancel endpoint when cancel clicked", async () => {
+    renderWithProviders(
+      <RouterProvider router={router} />,
+      {},
+      { wrapper: ReactQueryWrapper },
+      false
+    );
+
+    await waitFor(() => {
+      expect(screen.getByText("spotify.zip")).toBeInTheDocument();
+    });
+    const cancelBtn = screen.getByRole("button", { name: "Cancel import" });
+    await user.click(cancelBtn);
+  });
+
+  it("calls refresh endpoint when refresh clicked", async () => {
+    renderWithProviders(
+      <RouterProvider router={router} />,
+      {},
+      { wrapper: ReactQueryWrapper },
+      false
+    );
+
+    await waitFor(() => {
+      expect(screen.getByText("spotify.zip")).toBeInTheDocument();
+    });
+    const refreshBtn = screen.getByRole("button", { name: "Refresh" });
+    await user.click(refreshBtn);
```

Comment:
Same as above, there is no assertion here, it's a phantom test.
I think both of these last tests should have both a failure and a success tests (so 2 tests for each) that test the toast message shown on screen.

You can override the msw handlers with `server.use(...)` at the beginning of the test: https://mswjs.io/docs/best-practices/network-behavior-overrides